api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -116,7 +116,11 @@ public R or(R leftResult, R rightResult) {
     }
 
     public <T> R startsWith(BoundReference<T> ref, Literal<T> lit) {
-      throw new UnsupportedOperationException("Unsupported operation.");
+      throw new UnsupportedOperationException("startsWith operation is not supported by the visitor");
```

Comment:
nit: startsWitch expression

Reply:
So I noticed that a few of the other surrounding functions use `operation` as well. Namely, `IN` and `NOT_IN`, in addition to `STARTS_WITH` and now `NOT_STARTS_WITH`. I think it's likely because of throwing `UnsupportedOperationException`. I can update the others as well while I'm in here 👍

Reply:
I updated all of the ones I mentioned above in this file.